app/models/filters/filter_base.rb

```diff
@@ -465,7 +465,11 @@ def clean_comparison_pattern(pattern)
     end
 
     def available_coc_codes
-      GrdaWarehouse::Hud::ProjectCoc.distinct.pluck(:CoCCode, :hud_coc_code).flatten.map(&:presence).compact
+      @available_coc_codes ||= begin
+        cocs = GrdaWarehouse::Hud::ProjectCoc.distinct.pluck(:CoCCode, :hud_coc_code).flatten.map(&:presence).compact
+        # If a user has coc code limits assigned, enforce them
+        cocs & user&.coc_codes if user&.coc_codes.present?
```

Comment:
This may be desirable, but it occurs to me that this will hide any malformed CoC codes from the pick list

Reply:
It will, and I think that's desire-able.  I  It shouldn't hide them if you don't have a CoC Code limit.